pallets/crowdloans/src/crowdloan_structs.rs

```diff
@@ -0,0 +1,281 @@
+
+    /// Ask for a refund of the coins on the relay chain
+    fn refund<T: Config>(self, para_id: ParaId, currency: CurrencyId) -> DispatchResult;
+}
+
+impl<CurrencyId: std::fmt::Debug> ContributionStrategyExecutor<CurrencyId>
+    for ContributionStrategy<CurrencyId>
+{
+    fn execute<T: Config>(
+        self,
+        para_id: ParaId,
+        _currency_id: CurrencyId,
+        amount: BalanceOf<T>,
+    ) -> Result<(), DispatchError> {
+        let amount = amount.saturated_into::<u128>();
+
+        // TODO:
+        // determine correct fee amount
+        let fees = 900_000_000_000;
+
+        frame_support::ensure!(!fees.is_zero(), Error::<T>::XcmFeesCompensationTooLow);
+        let asset: MultiAsset = (MultiLocation::here(), fees).into();
+
+        let para_account = T::SelfParaId::get().into_account();
+        let derivative_index = T::DerivativeIndex::get();
+        let stash = T::DerivativeProvider::derivative_account_id(para_account, derivative_index);
+
+        let call = WestendCall::Utility(Box::new(UtilityCall::BatchAll(UtilityBatchAllCall {
+            calls: vec![
+                WestendCall::Balances::<T>(BalancesCall::TransferKeepAlive(
+                    BalancesTransferKeepAliveCall {
+                        dest: T::Lookup::unlookup(stash),
+                        value: amount,
+                    },
+                )),
+                WestendCall::Crowdloan(CrowdloanCall::Contribute(CrowdloanContributeCall::<T> {
+                    index: para_id,
+                    value: amount,
+                    signature: None,
+                })),
+            ],
+        })));
+
+        let xcm = Xcm(vec![
+            WithdrawAsset(MultiAssets::from(asset.clone())),
+            BuyExecution {
+                fees: asset.clone(),
+                weight_limit: Unlimited,
+            },
+            Transact {
+                origin_type: OriginKind::SovereignAccount,
+                // TODO:
+                // determine correct weighting for this calll
+                require_weight_at_most: 100_000_000_000,
+                call: call.encode().into(),
+            },
+            RefundSurplus,
+            DepositAsset {
+                assets: asset.into(),
+                max_assets: 1,
+                beneficiary: T::AccountIdToMultiLocation::convert(
+                    T::SelfParaId::get().into_account(),
+                ),
+            },
+        ]);
+
+        // TODO:
+        // correctly remove the tokens from our Parachain.
+        // this possibly applys to fees and the transfer amount
+        // burning on one side and creating on the other is called
+        // teleporting in the Polkadot world
+        // T::Assets::burn_from(101, &T::PalletId::get().into_account(), amount)?;
+
+        // send xcm call
+        let _response = T::XcmSender::send_xcm(MultiLocation::parent(), xcm)
+            .map_err(|_| Error::<T>::SendXcmError)?;
+
+        Ok(())
```

Comment:
This could be replaced with a simple `T::XcmSender::send_xcm(MultiLocation::parent(), xcm).map_err(|_| Error::<T>::SendXcmError)`